internal/pkg/composable/providers/kubernetes/hints_test.go

```diff
@@ -367,6 +367,149 @@ func TestGenerateHintsMappingWithLogStream(t *testing.T) {
 	assert.Equal(t, expected, hintsMapping)
 }
 
+func TestGenerateHintsMappingWithDefaultsandTypo(t *testing.T) {
+	logger := getLogger()
+	pod := &kubernetes.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod",
+			UID:       types.UID(uid),
+			Namespace: "testns",
+			Labels: map[string]string{
+				"foo":        "bar",
+				"with-dash":  "dash-value",
+				"with/slash": "some/path",
+			},
+			Annotations: map[string]string{
+				"app": "production",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		Spec: kubernetes.PodSpec{
+			NodeName: "testnode",
+		},
+		Status: kubernetes.PodStatus{PodIP: "127.0.0.5"},
+	}
+
+	mapping := map[string]interface{}{
+		"namespace": pod.GetNamespace(),
+		"pod": mapstr.M{
+			"uid":  string(pod.GetUID()),
+			"name": pod.GetName(),
+			"ip":   pod.Status.PodIP,
+		},
+		"namespace_annotations": mapstr.M{
+			"nsa": "nsb",
+		},
+		"labels": mapstr.M{
+			"foo":        "bar",
+			"with-dash":  "dash-value",
+			"with/slash": "some/path",
+		},
+		"annotations": mapstr.M{
+			"app": "production",
+		},
+	}
+	hints := mapstr.M{
+		"hints": mapstr.M{
+			"host":        "${kubernetes.pod.ip}:6379",
+			"package":     "redis",
+			"metricspath": "/metrics", // on purpose we have introduced a typo
```

Comment:
what is the typo here?

Reply:
Should be metrics_path

https://github.com/elastic/elastic-agent/blob/5a5e0f3f79a4b69c32762928417861c1ed033280/internal/pkg/composable/providers/kubernetes/hints.go#L26C17-L26C29

Reply:
for the purpose of the test - it is not important either it is a typo or not supported hint, maybe to make it obvious can be used smth like `"notSupportedHint"`?